### ISSUE TITLE
Known Issues with Colab Tutorials

### ISSUE THREAD

**Issue by rbharath** [deepchem/deepchem#1767]
I'm opening this issue to track known issues with running the tutorials on Colab and discussing potential solutions:

- Colab installation is slow: The tutorials currently install deepchem as in this [post](https://forum.deepchem.io/t/getting-deepchem-running-in-colab/81). The downside of this method is that it involves pulling down a full Anaconda python and installing DeepChem within. This is unfortunately needed since RDKit isn't available on pip
  - Potential Solution: We could possible swap to Miniconda to speed up installation times. I need to test this works on Colab stably though.
- Pngs don't display on some notebooks: Some of the tutorials, like the [solubility](https://github.com/deepchem/deepchem/blob/master/examples/tutorials/03_Modeling_Solubility.ipynb) example write png's to disk and display them in the Colab notebook. This method doesn't work on Colab for some reason. It's not clear what the right fix is here

**Comment by rbharath** [comment]
The issue with the `pngs` was fixed with using `display(Image(...)))` from Ipython. The colab installation speed still needs to be worked on...

**Comment by nd-02110114** [comment]
@rbharath 
When I replaced Anaconda to Miniconda and confirmed installation time, it changed about 6 min to 4 min, two min shorter. When I was doing a tutorial, the installation time could be very annoying...

Can I replace Anaconda with Miniconda?

**Comment by rbharath** [comment]
@nd-02110114 Yes, that would be a good contribution! Can you confirm that the notebooks install correctly on colab with miniconda? If so, PR would be welcome

**Comment by nd-02110114** [comment]
`01_The_Basic_Tools_of_the_Deep_Life_Sciences.ipynb` is no problem.  After I check all notebooks, I will submit PR!

**Comment by rbharath** [comment]
As a quick note, the Colab tutorials need to be modified to have the following magic command up top to enable TensorFlow 1.x on colab:

```
%tensorflow_version 1.x
```

**Comment by rbharath** [comment]
Copying over comments by @nd-02110114 from #1789. The following tutorials still have errors:

## Error details

- Tutorial Part 10: Exploring Quantum Chemistry with GDB1k
  - I think this tutorial is completely outdated. 
  - Path of `gdb1k.sdf` is broken. Although I used https://github.com/deepchem/deepchem/blob/master/datasets/gdb1k.sdf, the tutorials has still errors about dataset
- Tutorial Part 11: Learning Unsupervised Embeddings for Molecules
  - `model.fit_sequences(generate_sequences(40))` doesn't end...
- Tutorial Part 13: Modeling Protein-Ligand Interactions
  - Error at the line `import mdtraj as md`
  - Error Message `No module named 'mdtraj.formats.dcd' `
  - related issue : https://github.com/mdtraj/mdtraj/issues/1486
- Tutorial Part 18: Using Reinforcement Learning to Play Pong
  - Error at the line `env.env.render()`
  - Error Message `NoSuchDisplayException: Cannot connect to "None"`
- Tutorial Part 21: Modeling Protein-Ligand Interactions
  - Error at the line  `from Bio import SeqIO`
  - Error Message `ImportError: cannot import name '_aligners'`
  - related issue : https://github.com/biopython/biopython/issues/2809

**Comment by rbharath** [comment]
As a general comment, Tutorial 1 needs work. It's too down in the weeds when it should provide a higher level overview of the DeepChem project rather than immediately jump down into the weeds about splitters.

**Comment by rbharath** [comment]
Here's another error noted in #1793:
- Tutorial Part 19: Uses `dc.models.TensorGraph.load_from_dir('screen_model')` which uses the old TensorGraph style rather than the `model.restore()` keras style.

**Comment by nd-02110114** [comment]
I created deepchem install script for colab!
https://gist.github.com/nd-02110114/919a2d5a5f44992c1591e2c213208399

Install command became cleaner like blow.
```python
%tensorflow_version 1.x
!curl -Lo deepchem_installer.py https://git.io/JfVGS
import deepchem_installer
%time deepchem_installer.install(deepchem_version='2.3.0')
```
For future release
```python
!curl -Lo deepchem_installer.py https://git.io/JfVGS
import deepchem_installer
%time deepchem_installer.install(deepchem_version='2.4.0.rc')
```

And, I fixed two errors about importing third party module (`mdtraj` and `Bio`) https://github.com/deepchem/deepchem/issues/1767#issuecomment-605529882
I think Installing miniconda and conda packages to `/user/local/` is not so good.  Sometimes, package installation (especially includes native codes) fails because native codes don't compile correctly.

Currently, I know the PR which refactors tutorials will be merged soon. I just commented on this issue thread.

**Comment by paulsonak** [comment]
> Copying over comments by @nd-02110114 from #1789. 

> * Tutorial Part 11: Learning Unsupervised Embeddings for Molecules
>   
>   * `model.fit_sequences(generate_sequences(40))` doesn't end...

I stubbornly let this run. It does end (or maybe quits? I only saw up to checkpoint 30 even though it called for 40 epochs) on colab after 3+ hours, but it doesn't appear to work correctly. No losses or time elapsed are printed in the output and it predicts 0/500 smiles correctly. 

model.fit_sequences(generate_sequences(1)) takes ~5 minutes but has the same problems.

**Comment by rbharath** [comment]
I think we've addressed most of these issues (but of course new colab issues have arisen...) Closing this old issue as out of date
